### ISSUE TITLE
Failure to save Prefect blocks

### ISSUE THREAD

**Issue by jggatter** [anna-geller/dataflow-ops#7]
Hi,

When trying to run the Initial ECS Prefect Agent workflow, I'm encountering during the "Create Blocks & Upload to S3" step (`aws_creds.save(block_, overwrite=True)`) what seems to be a bad response:
```
Traceback (most recent call last):
  File "/home/runner/work/cyflows-infra/cyflows-infra/aws_ecs_blocks.py", line 17, in <module>
    aws_creds.save(block_, overwrite=True)
  File "/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/site-packages/prefect/utilities/asyncutils.py", line 230, in coroutine_wrapper
    return run_async_in_new_loop(async_fn, *args, **kwargs)
  File "/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/site-packages/prefect/utilities/asyncutils.py", line 181, in run_async_in_new_loop
    return anyio.run(partial(__fn, *args, **kwargs))
  File "/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/site-packages/anyio/_core/_eventloop.py", line 70, in run
    return asynclib.run(func, *args, **backend_options)
  File "/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/site-packages/anyio/_backends/_asyncio.py", line 292, in run
    return native_run(wrapper(), debug=debug)
  File "/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/asyncio/base_events.py", line 647, in run_until_complete
    return future.result()
  File "/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/site-packages/anyio/_backends/_asyncio.py", line 287, in wrapper
    return await func(*args)
  File "/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/site-packages/prefect/blocks/core.py", line 880, in save
    document_id = await self._save(name=name, overwrite=overwrite, client=client)
  File "/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/site-packages/prefect/client/utilities.py", line 47, in with_injected_client
    return await fn(*args, **kwargs)
  File "/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/site-packages/prefect/blocks/core.py", line 833, in _save
    await self.register_type_and_schema(client=client)
  File "/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/site-packages/prefect/client/utilities.py", line 47, in with_injected_client
    return await fn(*args, **kwargs)
  File "/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/site-packages/prefect/blocks/core.py", line 775, in register_type_and_schema
    block_type = await client.read_block_type_by_slug(
  File "/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/site-packages/prefect/client/orion.py", line 1075, in read_block_type_by_slug
    return BlockType.parse_obj(response.json())
  File "/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/site-packages/httpx/_models.py", line 7[56](https://github.com/Cellarity/cyflows-infra/actions/runs/4109283325/jobs/7091605718#step:7:57), in json
    return jsonlib.loads(self.text, **kwargs)
  File "/opt/hostedtoolcache/Python/3.9.16/x[64](https://github.com/Cellarity/cyflows-infra/actions/runs/4109283325/jobs/7091605718#step:7:65)/lib/python3.9/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```
I ran the code of this aws_blocks.py script in my local REPL (with the same prefect config) and it worked successfully. However, I plan to use something like this step in my own workflows so I want to make sure that these calls to Prefect can be made from GitHub Actions in my organization's private repository.

Further info from the logs:
```python
  id_ = "***"
  key_ = "***"
  path_ = "cyflows/test"
  img_ = "<AWS_ACCOUNT_ID>.dkr.ecr.us-east-1.amazonaws.com/cyflows:b07720be5ad89c0e1dd7675fcb545df43c8148c9".  # Redacted account ID
  block_ = "$BLOCK"
  cluster_ = "prefect"
  cpu_ = "512"
  memory_ = "1024"
  aws_acc_id = "$AWS_ACCOUNT_ID"
  exec_role = f"arn:aws:iam::{aws_acc_id}:role/cyflows_ecs_execution_role"
  task_role = f"arn:aws:iam::{aws_acc_id}:role/cyflows_ecs_execution_role"
```

Previous steps:
```
Run prefect config set PREFECT_API_KEY=***
  prefect config set PREFECT_API_KEY=***
  prefect config set PREFECT_API_URL=***
  shell: /usr/bin/bash -e {0}
  env:
    PROJECT: cyflows
    ECS_CLUSTER: prefect
    BLOCK: cyflows-test
    AWS_REGION: us-east-1
    PREFECT_VERSION: 2.*
    pythonLocation: /opt/hostedtoolcache/Python/3.9.16/x64
    PKG_CONFIG_PATH: /opt/hostedtoolcache/Python/3.9.16/x64/lib/pkgconfig
    Python_ROOT_DIR: /opt/hostedtoolcache/Python/3.9.16/x64
    Python2_ROOT_DIR: /opt/hostedtoolcache/Python/3.9.16/x64
    Python3_ROOT_DIR: /opt/hostedtoolcache/Python/3.9.16/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.9.16/x64/lib
Set 'PREFECT_API_KEY' to '***'.
Updated profile 'default'.
Set 'PREFECT_API_URL' to 
'https://app.prefect.cloud/account/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/workspac
e/15d830f1-346b-411c-9949-2f61c0e470ab'.  # Redacted account/organization ID
Updated profile 'default'.
```
```
Run aws-actions/configure-aws-credentials@v1
  with:
    aws-access-key-id: ***
    aws-secret-access-key: ***
    aws-region: us-east-1
    mask-aws-account-id: no
    audience: sts.amazonaws.com
  env:
    PROJECT: cyflows
    ECS_CLUSTER: prefect
    BLOCK: cyflows-test
    AWS_REGION: us-east-1
    PREFECT_VERSION: 2.*
    pythonLocation: /opt/hostedtoolcache/Python/3.9.16/x64
    PKG_CONFIG_PATH: /opt/hostedtoolcache/Python/3.9.16/x64/lib/pkgconfig
    Python_ROOT_DIR: /opt/hostedtoolcache/Python/3.9.16/x64
    Python2_ROOT_DIR: /opt/hostedtoolcache/Python/3.9.16/x64
    Python3_ROOT_DIR: /opt/hostedtoolcache/Python/3.9.16/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.9.16/x64/lib

Warning: The `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
```

Pip seems to be grabbing:
prefect-2.7.12
prefect_aws-0.2.4

Where might I be going wrong? Any insights would be very appreciated!

**Comment by jggatter** [comment]
When I create the blocks via running the script locally and manually upload the `infrastructure/ecr_repository.yml` and `infrastructure/ecs_cluster_prefect_agent.yml` templates directly to CloudFormation, my Prefect Agent seems to also encounter this same problem:
```
Starting v2.7.12 agent connected to 
https://app.prefect.cloud/account/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/workspace
/15d830f1-346b-411c-9949-2f61c0e470ab...
  ___ ___ ___ ___ ___ ___ _____     _   ___ ___ _  _ _____
 | _ \ _ \ __| __| __/ __|_   _|   /_\ / __| __| \| |_   _|
 |  _/   / _|| _|| _| (__  | |    / _ \ (_ | _|| .` | | |
 |_| |_|_\___|_| |___\___| |_|   /_/ \_\___|___|_|\_| |_|
Agent started! Looking for work from queue(s): cyflows...
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/prefect/cli/_utilities.py", line 41, in wrapper
    return fn(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/prefect/utilities/asyncutils.py", line 230, in coroutine_wrapper
    return run_async_in_new_loop(async_fn, *args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/prefect/utilities/asyncutils.py", line 181, in run_async_in_new_loop
    return anyio.run(partial(__fn, *args, **kwargs))
  File "/usr/local/lib/python3.9/site-packages/anyio/_core/_eventloop.py", line 70, in run
    return asynclib.run(func, *args, **backend_options)
  File "/usr/local/lib/python3.9/site-packages/anyio/_backends/_asyncio.py", line 292, in run
    return native_run(wrapper(), debug=debug)
  File "/usr/local/lib/python3.9/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/local/lib/python3.9/asyncio/base_events.py", line 647, in run_until_complete
    return future.result()
  File "/usr/local/lib/python3.9/site-packages/anyio/_backends/_asyncio.py", line 287, in wrapper
    return await func(*args)
  File "/usr/local/lib/python3.9/site-packages/prefect/cli/agent.py", line 192, in start
    tg.start_soon(
  File "/usr/local/lib/python3.9/site-packages/anyio/_backends/_asyncio.py", line 662, in __aexit__
    raise exceptions[0]
  File "/usr/local/lib/python3.9/site-packages/prefect/utilities/services.py", line 46, in critical_service_loop
    await workload()
  File "/usr/local/lib/python3.9/site-packages/prefect/agent.py", line 200, in get_and_submit_flow_runs
    async for work_queue in self.get_work_queues():
  File "/usr/local/lib/python3.9/site-packages/prefect/agent.py", line 143, in get_work_queues
    work_queue = await self.client.read_work_queue_by_name(
  File "/usr/local/lib/python3.9/site-packages/prefect/client/orion.py", line 793, in read_work_queue_by_name
    return schemas.core.WorkQueue.parse_obj(response.json())
  File "/usr/local/lib/python3.9/site-packages/httpx/_models.py", line 756, in json
    return jsonlib.loads(self.text, **kwargs)
  File "/usr/local/lib/python3.9/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "/usr/local/lib/python3.9/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/local/lib/python3.9/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
An exception occurred.
```
The only differences between what I'm doing and the workflow does is that 
1) I don't have any Prefect deployments at this point
2) the ECS cluster was created to have the `prefecthq/prefect:2-python3.9` image since that was the default.

Another detail is that the API key used belongs to a Prefect Service account which was added to the workspace with Developer privileges.

**Comment by jggatter** [comment]
```
{
    "taskDefinitionArn": "arn:aws:ecs:us-east-1:xxxxxxxxxxxx:task-definition/cyflows:3",
    "containerDefinitions": [
        {
            "name": "cyflows",
            "image": "xxxxxxxxxxxx.dkr.ecr.us-east-1.amazonaws.com/cyflows:b07720be5ad89c0e1dd7675fcb545df43c8148c9",
            "cpu": 0,
            "links": [],
            "portMappings": [],
            "essential": true,
            "entryPoint": [
                "bash",
                "-c"
            ],
            "command": [
                "prefect agent start -q cyflows"
            ],
            "environment": [
                {
                    "name": "AWS_RETRY_MODE",
                    "value": "adaptive"
                },
                {
                    "name": "AWS_MAX_ATTEMPTS",
                    "value": "10"
                },
                {
                    "name": "PREFECT_LOGGING_LEVEL",
                    "value": "INFO"
                }
            ],
            "environmentFiles": [],
            "mountPoints": [],
            "volumesFrom": [],
            "secrets": [
                {
                    "name": "PREFECT_API_URL",
                    "valueFrom": "arn:aws:ssm:us-east-1:xxxxxxxxxxxx:parameter/PREFECT_API_URL"
                },
                {
                    "name": "PREFECT_API_KEY",
                    "valueFrom": "arn:aws:ssm:us-east-1:xxxxxxxxxxxx:parameter/PREFECT_API_KEY"
                }
            ],
            "stopTimeout": 120,
            "dnsServers": [],
            "dnsSearchDomains": [],
            "extraHosts": [],
            "dockerSecurityOptions": [],
            "dockerLabels": {},
            "ulimits": [],
            "logConfiguration": {
                "logDriver": "awslogs",
                "options": {
                    "awslogs-group": "cyflows",
                    "awslogs-region": "us-east-1",
                    "awslogs-stream-prefix": "cyflows"
                },
                "secretOptions": []
            },
            "systemControls": []
        }
    ],
    "family": "cyflows",
    "taskRoleArn": "arn:aws:iam::xxxxxxxxxxxx:role/cyflows_ecs_task_role",
    "executionRoleArn": "arn:aws:iam::xxxxxxxxxxxx:role/cyflows_ecs_execution_role",
    "networkMode": "awsvpc",
    "revision": 3,
    "volumes": [],
    "status": "ACTIVE",
    "requiresAttributes": [
        {
            "name": "com.amazonaws.ecs.capability.logging-driver.awslogs"
        },
        {
            "name": "ecs.capability.execution-role-awslogs"
        },
        {
            "name": "com.amazonaws.ecs.capability.ecr-auth"
        },
        {
            "name": "com.amazonaws.ecs.capability.docker-remote-api.1.19"
        },
        {
            "name": "com.amazonaws.ecs.capability.docker-remote-api.1.17"
        },
        {
            "name": "com.amazonaws.ecs.capability.task-iam-role"
        },
        {
            "name": "ecs.capability.container-ordering"
        },
        {
            "name": "ecs.capability.execution-role-ecr-pull"
        },
        {
            "name": "ecs.capability.secrets.ssm.environment-variables"
        },
        {
            "name": "com.amazonaws.ecs.capability.docker-remote-api.1.18"
        },
        {
            "name": "ecs.capability.task-eni"
        }
    ],
    "placementConstraints": [],
    "compatibilities": [
        "EC2",
        "FARGATE"
    ],
    "requiresCompatibilities": [
        "FARGATE"
    ],
    "cpu": "512",
    "memory": "1024",
    "registeredAt": "2023-02-07T13:58:24.489000-05:00",
    "registeredBy": "arn:aws:sts::xxxxxxxxxxxx:assumed-role/development-admin/xxxxxxxxx@xxxxxxxx.com"
}
```
Reuploaded the template with the built image of the repo, but no luck. Here's the task definition.

**Comment by anna-geller** [comment]
I looked a bit and I couldn't reproduce on the latest version of this repo on main branch -- you might want to run first the action nr 3 to delete all old resources and then action nr 1 to recreate all from scratch

it might be some python-action/caching issue too. you could try leveraging a higher python version and adding a cache for pip:
```
  steps:
    - name: Set up Python
      uses: actions/setup-python@v4
      with:
        python-version: "3.10"
        cache: 'pip'
```

lastly, you can try this approach when you create blocks in python and call with argparse as here:  https://github.com/anna-geller/prefect-cloud-gcp/tree/main/.github/actions/blocks-quickstart 

https://github.com/anna-geller/prefect-cloud-gcp/blob/main/.github/actions/blocks-quickstart/action.yml#L54-L55

**Comment by jggatter** [comment]
As you suggested, I tried from scratch. I realized that as I pasted the URL into the `PREFECT_API_URL` secret it was being pasted with newlines, e.g.
```
https://app.prefect.cloud/account/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/workspac
e/15d830f1-346b-411c-9949-2f61c0e470ab
```
Tried again with setting the URL secret again in the old setup and it worked. Whitespace strikes again. I thought it was just wordwrap, I can't believe it...
It might be a good idea for the API to do some input validation to make sure the the URL is in a valid format.

Thanks for your help, Anna!
